Core/Services/ConditionsLogsService.cs

```diff
@@ -91,18 +92,18 @@ private RequirementLevel CalculateHumidityLevel (double value)
         };
     }
     
-    private int CalculateMood (ConditionsLog conditionsLog)
+    private async Task<int> CalculateMood (ConditionsLog conditionsLog)
     {
         // Compare ideal requirements for humidity, temperature, soil moisture and light level with actual conditions and calculate mood from 0-4
         // get ideal requirements from plant
-        var requirementsForPlant = plantRepository.GetRequirementsForPlant(conditionsLog.PlantId);
+        var requirementsForPlant = await requirementService.GetRequirements(conditionsLog.PlantId);
         // compare with actual conditions
         var mood = 0;
         // calculate mood
-        mood += CalculateScore((int)requirementsForPlant.Result.HumidityLevel, (int)CalculateHumidityLevel(conditionsLog.Humidity));
-        mood += CalculateScore((int)requirementsForPlant.Result.TemperatureLevel, (int)CalculateTemperatureLevel(conditionsLog.Temperature));
-        mood += CalculateScore((int)requirementsForPlant.Result.SoilMoistureLevel, (int)CalculateSoilMoistureLevel(conditionsLog.SoilMoisture));
-        mood += CalculateScore((int)requirementsForPlant.Result.LightLevel, (int)CalculateLightLevel(conditionsLog.Light));
```

Comment:
why remove the result?

Reply:
it is not needed, why not await the task and work directly with the requirements class, rather than the Task<Requirements> wrapper, which requires us to access properties through result

Reply:
ah yes i see, that works